core/capabilities.go

```diff
@@ -543,13 +542,6 @@ func NewCapabilities(caps []Capability, m []Capability) *Capabilities {
 	return c
 }
 
-func NewCapabilitiesWithConstraints(caps []Capability, m []Capability, constraints Constraints, capabilityConstraints CapabilityConstraints) *Capabilities {
```

Comment:
@Yondonfu, just double-checking if there was a specific reason for initializing the constraints in the constructor?

Reply:
Maybe because of `nil` errors 🤔. I will do some debugging to see if I can get this to work.

Reply:
> just double-checking if there was a specific reason for initializing the constraints in the constructor?

IIRC didn't have much of a reason outside of the fact at the time constraints were only set once upon initializing a Capabilities struct (i.e. based on the models configuration file on startup for an orchestrator or based on the model requested by a gateway while discovering orchestrators for selection) so I think I opted to just set the constraints once in the constructor leaving changes to support modification of the constraints to be made later on when there was a better sense of how constraints should be updated over time (i.e. to reflect changes in supported models).